Zelda/Zelda.cpp

```diff
@@ -1,1205 +0,0 @@
-{
-	return name;
-}
-
-
-
-//---------------------Princess------------------------
-
-
-
-Princess::Princess()
-{
-	is_Alive = true;
-}
-
-
-
-Princess::~Princess()
-{
-
-}
-
-
-
-bool Princess::getLivingState()
-{
-	return is_Alive;
-}
-
-
-
-void Princess::setLivingState(bool currentState)
-{
-	is_Alive = currentState;
-}
-
-
-
-//---------------------Room------------------------
-
-
-
-Room::Room()
-{
-	for (int i = 0; i < 4; ++i)
-	{
-		paths[i] = nullptr;
-	}
-
-	for (int i = 0; i < 5; ++i)
-	{
-		items_Present[i] = nullptr;
-	}
-
-	monster_Present  = nullptr;
-	princess_Present = nullptr;
-
-	description 	 = nullptr;
-	roomNumber 		 = 0;
-}
-
-
-
-Room::~Room()
-{
-	if(description)
-		delete[] description;
-}
-
-
-
-int Room::directionIndex(char const* direction)
-{
-	int i = 0;
-	if(strcmp(direction , "NORTH") == 0)
-		i = 0;
-
-	else if(strcmp (direction, "SOUTH") == 0)
-		i = 1;
-
-	else if(strcmp (direction, "EAST") == 0)
-		i = 2;
-
-	else if(strcmp (direction, "WEST") == 0)
-		i = 3;
-
-	return i;
-}
-
-
-
-char* Room::directionName(int i)
-{
-	char* buffer = new char [10];
-
-	if( i == 0)
-		strcpy_s(buffer,10, "North");
-
-	else if(i == 1)
-		strcpy_s(buffer,10, "South");
-
-	else if(i == 2)
-		strcpy_s(buffer,10, "East");
-
-	else if(i == 3)
-		strcpy_s(buffer,10, "West");
-
-	return buffer;
-}
-
-
-
-bool Room::isRoomFull()
-{
-	for (int i = 0; i < 5; ++i)
-	{
-		if (items_Present[i] == nullptr)
-		{
-			return false;
-		}
-	}
-
-	return true;
-}
-
-
-
-//Getters
-
-
-
-Room** Room::getPaths()
-{
-	return paths;
-}
-
-
-
-Item** Room::getItemsPresent()
-{
-	return items_Present;
-}
-
-
-
-Monster* Room::getMonsterPresent()
-{
-	return monster_Present;
-}
-
-
-
-Princess* Room::getPrincessPresent()
-{
-	return princess_Present;
-}
-
-
-
-char* Room::getDescription()
-{
-	return description;
-}
-
-
-
-int Room::getRoomNumber()
-{
-	return roomNumber;
-}
-
-
-
-
-// paths[0] = North
-// paths[1] = South
-// paths[2] = East
-// paths[3] = West
-
-
-
-void Room::setNorth(Room* roomNo)
-{
-	paths[0] = roomNo;
-}
-
-
-
-void Room::setSouth(Room* roomNo)
-{
-	paths[1] = roomNo;
-}
-
-
-
-void Room::setEast(Room* roomNo)
-{
-	paths[2] = roomNo;
-}
-
-
-
-void Room::setWest(Room* roomNo)
-{
-	paths[3] = roomNo;
-}
-
-
-
-void Room::setItemsPresent(int itemNo , Item* item)
-{
-	items_Present[itemNo] = item;
-}
-
-
-
-void Room::setMonsterPresent(Monster* monster)
-{
-	monster_Present = monster;
-}
-
-
-
-void Room::setPrincessPresent(Princess* princess)
-{
-	princess_Present = princess;
-}
-
-
-
-void Room::setDesription(char const* room_Description)
-{
-	description = new char [strlen(room_Description)+1];
-	strcpy_s(description,strlen(room_Description)+1,room_Description);
-}
-
-
-
-void Room::setRoomNumber(int roomNo)
-{
-	roomNumber = roomNo;
-}
-
-
-
-//---------------------Castle------------------------
-
-
-
-Castle::Castle()
-{
-
-}
-
-
-
-Room* Castle::getRoom( int room_No )
-{
-	return &rooms[room_No - 1];
```

Comment:
room_No - 1 could be outside the range of rooms[].  Add a bounds check to ensure it is no out of bounds.